### PR TITLE
[MM-44159] Add user sort by admin status in channels

### PR DESCRIPTION
**Pull request by JulienTant**
#### Summary
Allows us to sort users of a channel by admin. The channel admins will be returned first, followed by all other members.

#### Frontend PR

https://github.com/mattermost/mattermost-webapp/pull/10577

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-44159

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: store/sqlstore/user_store.go [hunk 1]
```diff
@@ -771,6 +771,37 @@ func (us SqlUserStore) GetProfilesInChannelByStatus(options *model.UserGetOption
...
+
+	if options.Inactive && !options.Active {
+		query = query.Where("u.DeleteAt != 0")
+	} else if options.Active && !options.Inactive {
+		query = query.Where("u.DeleteAt = 0")
+	}
+
+	queryString, args, err := query.ToSql()
+	if err != nil {
+		return nil, errors.Wrap(err, "get_profiles_in_channel_by_admin_tosql")
```
Comment (pvev) [inline store/sqlstore/user_store.go:790]: is this expected to be translated somehow?

Comment (JulienTant) [reply store/sqlstore/user_store.go:790]: Nop those are for logging purpose

**Review by pvev — APPROVED** [review]
LGTM! thanks @JulienTant :shipit:

File: store/sqlstore/user_store.go (hunk 1 above)
Comment (pvev) [reply store/sqlstore/user_store.go:790]: ok, question then, why the snake case format for the log? wouldn't it be better to write a normal text to log?

**Pull request merged**